Rename of docs

SwartzCr opened the pull request:
some rename of .rst files

I tried to be smart about just renaming references to the client, but I've also changed some urls to certbot/certbot which will have to wait for the GH repo to be moved

SwartzCr commented:
@bmw  wanna take a look over this?

bmw commented on docs/api/cb_util.rst:
```diff
@@ -0,0 +1,5 @@
+:mod:`certbot.cb_util`
```
This file should be deleted as we didn't rename `le_util` to `cb_util` to prevent breakage.

bmw commented on docs/api/le_util.rst:
```diff
@@ -1,5 +0,0 @@
-:mod:`certbot.le_util`
```
Similarly, this file should be added again.

bmw commented on docs/ciphers.rst:
```diff
@@ -156,9 +153,9 @@ recommendations with sources of expert guidance on ciphersuites and other
 suggestions. The recommendations we received are available at
 
-https://github.com/letsencrypt/letsencrypt/wiki/Ciphersuite-guidance
+https://github.com/certbot/certbot/wiki/Ciphersuite-guidance
```
Do we want to start changing these URLs yet? If we merge this now, our documentation will be broken.

We can either:

1. Wait to merge this PR until we move the repo.
2. Make these changes in a later PR, perhaps even after the 0.6.0 release. Leaving references to `https://github.com/letsencrypt/letsencrypt/*` shouldn't break anything.

I'm personally in favor of 2.

bmw commented on docs/contributing.rst:
```diff
@@ -146,7 +146,7 @@ encouraged) make sure to install prerequisites as listed in
 the integration tests suite.
 
-.. _Boulder: https://github.com/letsencrypt/boulder
+.. _Boulder: https://github.com/certbot/boulder
```
This will remain https://github.com/letsencrypt/boulder

bmw commented on docs/using.rst:
```diff
@@ -25,33 +25,33 @@ To install and run the client, just type...
+   ./certbot-auto
 
-.. hint:: During the beta phase, Let's Encrypt enforces strict rate limits on
+.. hint:: During the beta phase, Certbot enforces strict rate limits on
```
This is referring to the Let's Encrypt server and should not be changed.

bmw commented on docs/using.rst:
```diff
@@ -93,10 +93,10 @@ s3front_    Y    Y    Integration with Amazon CloudFront distribution of S3 buck
+.. _plesk: https://github.com/plesk/certbot-plesk
+.. _haproxy: https://code.greenhost.net/open/certbot-haproxy
+.. _s3front: https://github.com/dlapiduz/certbot-s3front
+.. _gandi: https://github.com/Gandi/certbot-gandi
```
We should not change any of these URLs. We do not know when/if they'll be renamed or what they'll be renamed to.

bmw commented on docs/using.rst:
```diff
@@ -130,21 +130,21 @@ specified ``--webroot-path``.  So, for instance,
-domains in ``${webroot-path}/.well-known/acme-challenge``. Then the Let's
-Encrypt validation server makes HTTP requests to validate that the DNS for each
-requested domain resolves to the server running letsencrypt. An example request
+domains in ``${webroot-path}/.well-known/acme-challenge``. Then the Certbot
```
Reference to Let's Encrypt server

bmw commented on docs/using.rst:
```diff
@@ -130,21 +130,21 @@ specified ``--webroot-path``.  So, for instance,
 ::
 
-    66.133.109.36 - - [05/Jan/2016:20:11:24 -0500] "GET /.well-known/acme-challenge/HGr8U1IeTW4kY_Z6UIyaakzOkyQgPr_7ArlLgtZE8SX HTTP/1.1" 200 87 "-" "Mozilla/5.0 (compatible; Let's Encrypt validation server; +https://www.letsencrypt.org)"
+    66.133.109.36 - - [05/Jan/2016:20:11:24 -0500] "GET /.well-known/acme-challenge/HGr8U1IeTW4kY_Z6UIyaakzOkyQgPr_7ArlLgtZE8SX HTTP/1.1" 200 87 "-" "Mozilla/5.0 (compatible; Certbot validation server; +https://www.certbot.com)"
```
Request will still come from Let's Encrypt server.

bmw commented on docs/using.rst:
```diff
@@ -272,22 +272,22 @@ you prefer to manage everything by hand, this section provides
+.. note:: ``/etc/certbot/archive`` and ``/etc/certbot/keys``
    contain all previous keys and certificates, while
-   ``/etc/letsencrypt/live`` symlinks to the latest versions.
+   ``/etc/certbot/live`` symlinks to the latest versions.
```
All `/etc/certbot` paths above are incorrect. I believe we decided not to make this change now (or maybe ever). See #2830.

bmw commented on docs/using.rst:
```diff
@@ -340,17 +340,17 @@ Configuration file
-  ``~/.config/letsencrypt/cli.ini`` if ``$XDG_CONFIG_HOME`` is not
+- ``/etc/certbot/cli.ini``
+- ``$XDG_CONFIG_HOME/certbot/cli.ini`` (or
+  ``~/.config/certbot/cli.ini`` if ``$XDG_CONFIG_HOME`` is not
```
Path is still `~/.config/letsencrypt/cli.ini`.

bmw commented on docs/using.rst:
```diff
@@ -359,21 +359,21 @@ By default, the following locations are searched:
-Freenode) <https://webchat.freenode.net?channels=%23letsencrypt>`_ or
-get support on our `forums <https://community.letsencrypt.org>`_.
+If you're having problems you can chat with us on `IRC (#certbot @
+OFTC) <https://webchat.oftc.net?channels=%23certbot>`_ or
```
Are we moving to OFTC? I'm fine either way, however, @pde and I are hanging out in #certbot on Freenode.

SwartzCr replied:
@pde and I are also hanging out in #certbot in OFTC and he mentioned on a call that he preferred OFTC to freenode (for this?)

bmw commented on docs/using.rst:
```diff
@@ -359,21 +359,21 @@ By default, the following locations are searched:
-get support on our `forums <https://community.letsencrypt.org>`_.
+If you're having problems you can chat with us on `IRC (#certbot @
+OFTC) <https://webchat.oftc.net?channels=%23certbot>`_ or
+get support on our `forums <https://community.certbot.org>`_.
```
This should still be community.letsencrypt.org.

bmw commented on docs/using.rst:
```diff
@@ -359,21 +359,21 @@ By default, the following locations are searched:
   that the latter might include some personally identifiable
   information, including your email and domains)
-- copy and paste logs from ``/var/log/letsencrypt`` (though mind they
+- copy and paste logs from ``/var/log/certbot`` (though mind they
```
Incorrect path

bmw commented on docs/using.rst:
```diff
@@ -402,14 +402,14 @@ to, `install Docker`_, then issue the following command:
+   sudo docker run -it --rm -p 443:443 -p 80:80 --name certbot \
+               -v "/etc/certbot:/etc/certbot" \
+               -v "/var/lib/certbot:/var/lib/certbot" \
+               quay.io/certbot/certbot:latest auth
```
All `-v` arguments are incorrect. Additionally, we should probably keep the quay.io URL as `letsencrypt/letsencrypt`, at least for the time being.

bmw commented on docs/using.rst:
```diff
@@ -402,14 +402,14 @@ to, `install Docker`_, then issue the following command:
 and follow the instructions (note that ``auth`` command is explicitly
 used - no installer plugins involved). Your new cert will be available
-in ``/etc/letsencrypt/live`` on the host.
+in ``/etc/certbot/live`` on the host.
```
Bad path

bmw commented on docs/using.rst:
```diff
@@ -473,7 +473,7 @@ does include the nginx plugin package:
    layman -S
    layman -a mrueg
-   emerge -av app-crypt/letsencrypt-nginx
+   emerge -av app-crypt/certbot-nginx
```
Like 3rd party plugins, these instructions should not be changed until we know when/if/how they will be changed.

bmw commented on docs/using.rst:
```diff
@@ -10,12 +10,12 @@ User Guide
 ============
 
-.. _letsencrypt-auto:
+.. _certbot-auto:
```
Similarly, `certbot-auto` doesn't exist yet. We can either wait to merge this or revert these changes for the time being.

bmw commented:
I did not comment on all GitHub URLs or references to `certbot-auto`. I also found these references to Let's Encrypt which should be changed:
```
ciphers.rst:75:Let's Encrypt project hopes to provide useful defaults that reflect
ciphers.rst:120:The Let's Encrypt project expects to follow the Mozilla recommendations
ciphers.rst:130:The Let's Encrypt project may deviate from the Mozilla recommendations
ciphers.rst:134:team first, before asking the Let's Encrypt project to change
ciphers.rst:139:The Let's Encrypt project will entertain proposals to create a *very*
```

`README.rst` was also not modified. Since #2869 was created, we can potentially wait on this, however, since this file is also used in our readthedocs documentation (see `docs/intro.rst`), this PR should not be merged until the README is updated as well.

Finally, I found references to Let's Encrypt in some of the documentation build scripts. I don't know much about this stuff but maybe they should be changed in well. I found:
```
Makefile:89:    @echo "# qcollectiongenerator $(BUILDDIR)/qthelp/LetsEncrypt.qhcp"
Makefile:91:    @echo "# assistant -collectionFile $(BUILDDIR)/qthelp/LetsEncrypt.qhc"
Makefile:98:    @echo "# mkdir -p $$HOME/.local/share/devhelp/LetsEncrypt"
Makefile:99:    @echo "# ln -s $(BUILDDIR)/devhelp $$HOME/.local/share/devhelp/LetsEncrypt"
conf.py:67:project = u'Let\'s Encrypt'
conf.py:68:copyright = u'2014-2015, Let\'s Encrypt Project'
conf.py:228:htmlhelp_basename = 'LetsEncryptdoc'
conf.py:250:    ('index', 'LetsEncrypt.tex', u'Let\'s Encrypt Documentation',
conf.py:251:     u'Let\'s Encrypt Project', 'manual'),
conf.py:280:    ('index', 'certbot', u'Let\'s Encrypt Documentation',
conf.py:296:    ('index', 'LetsEncrypt', u'Let\'s Encrypt Documentation',
conf.py:297:     u'Let\'s Encrypt Project', 'LetsEncrypt', 'One line description of project.',
make.bat:130:   echo.^> qcollectiongenerator %BUILDDIR%\qthelp\LetsEncrypt.qhcp
make.bat:132:   echo.^> assistant -collectionFile %BUILDDIR%\qthelp\LetsEncrypt.ghc
```

SwartzCr commented:
added README.rst, changed all github urls back to LE, but left in references to certbot-auto, so this should be merged only after certbot-auto has been authored

jsha commented on README.rst:
```diff
@@ -3,18 +3,18 @@
 `Frequently Asked Questions (FAQ) <https://community.letsencrypt.org/t/frequently-asked-questions-faq/26#topic-title>`_.
 
-About the Let's Encrypt Client
+About Certbot
```
Add a second paragraph saying "Until May 2016, Certbot was named simply ``letsencrypt`` or ``letsencrypt-auto``, depending on install method. Instructions on the Internet may still refer to this older name."

jsha commented on docs/ciphers.rst:
```diff
@@ -80,30 +78,29 @@ art. However, the Let's Encrypt certificate authority does *not*
-if you obtain a certificate using *standalone* mode and then manually
-install it in an IMAP or LDAP server, your cryptographic settings will
-not be modified by the client in any way.
+If you don't use Certbot to configure your server directly, because the 
```
I would recommend against reflowing text, unless there are really egregious line length issues, since it makes `git blame` less useful.

jsha commented on docs/contributing.rst:
```diff
@@ -175,8 +175,8 @@ Configurators may implement just one of those).
 
-.. _interfaces.py: https://github.com/letsencrypt/letsencrypt/blob/master/certbot/interfaces.py
-.. _plugins/common.py: https://github.com/letsencrypt/letsencrypt/blob/master/certbot/plugins/common.py#L34
+.. _interfaces.py: https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/interfaces.py
```
This change is incorrect.

jsha commented on docs/contributing.rst:
```diff
@@ -175,8 +175,8 @@ Configurators may implement just one of those).
-.. _interfaces.py: https://github.com/letsencrypt/letsencrypt/blob/master/certbot/interfaces.py
-.. _plugins/common.py: https://github.com/letsencrypt/letsencrypt/blob/master/certbot/plugins/common.py#L34
+.. _interfaces.py: https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/interfaces.py
+.. _plugins/common.py: https://github.com/letsencrypt/letsencrypt/blob/master/letsencrypt/plugins/common.py#L34
```
This one too.

jsha commented on docs/packaging.rst:
```diff
@@ -3,4 +3,4 @@ Packaging Guide
 
 Documentation can be found at
-https://github.com/letsencrypt/letsencrypt/wiki/Packaging.
+https://github.com/certbot/certbot/wiki/Packaging.
```
This change is premature until the repo rename.

jsha commented on docs/using.rst:
```diff
@@ -25,33 +25,33 @@ To install and run the client, just type...
-.. hint:: During the beta phase, Let's Encrypt enforces strict rate limits on
-   the number of certificates issued for one domain. It is recommended to
-   initially use the test server via `--test-cert` until you get the desired
+.. hint:: During the beta phase, the Let's Encrypt servers enforce strict rate
```
While we're in here, can we remove "During the beta phase," and "strict?"

jsha commented on docs/using.rst:
```diff
@@ -130,21 +130,21 @@ specified ``--webroot-path``.  So, for instance,
 ::
 
-    66.133.109.36 - - [05/Jan/2016:20:11:24 -0500] "GET /.well-known/acme-challenge/HGr8U1IeTW4kY_Z6UIyaakzOkyQgPr_7ArlLgtZE8SX HTTP/1.1" 200 87 "-" "Mozilla/5.0 (compatible; Let's Encrypt validation server; +https://www.letsencrypt.org)"
+    66.133.109.36 - - [05/Jan/2016:20:11:24 -0500] "GET /.well-known/acme-challenge/HGr8U1IeTW4kY_Z6UIyaakzOkyQgPr_7ArlLgtZE8SX HTTP/1.1" 200 87 "-" "Mozilla/5.0 (compatible; Let's Encryptvalidation server; +https://www.letsencrypt.org)"
```
Missing space.

jsha commented on docs/using.rst:
```diff
@@ -359,8 +359,8 @@ By default, the following locations are searched:
 
-If you're having problems you can chat with us on `IRC (#letsencrypt @
-Freenode) <https://webchat.freenode.net?channels=%23letsencrypt>`_ or
+If you're having problems you can chat with us on `IRC (#certbot @
```
I think we're keeping #letsencrypt on Freenode for now.

jsha commented on docs/using.rst:
```diff
@@ -420,7 +420,7 @@ Operating System Packages
 **FreeBSD**
 
-  * Port: ``cd /usr/ports/security/py-letsencrypt && make install clean``
+  * Port: ``cd /usr/ports/security/py-letsencrypt make install clean``
```
Missing &&

jsha commented:
LGTM

SwartzCr commented on docs/ciphers.rst:
```diff
@@ -127,15 +124,15 @@ which uses the ChaCha and Poly1305 algorithms, and which is already
+that point, Certbot would likely follow the Mozilla recommendations and favor
+the use of this ciphersuite as well.)
 
 The Let's Encrypt project may deviate from the Mozilla recommendations
```
@jsha what do you think about the use of LEt's Encrypt in this paragraph and the next?

jsha replied:
You mean this paragraph and the previous? I think Certbot is more accurate here.

schoen commented on certbot-compatibility-test/docs/conf.py:
```diff
@@ -311,7 +311,7 @@
-    'certbot': ('https://letsencrypt.readthedocs.org/en/latest/', None),
+    'certbot': ('https://certbot.eff.org/docs/', None),
     'certbot-apache': (
         'https://letsencrypt-apache.readthedocs.org/en/latest/', None),
```
How come this one didn't change?

schoen commented on docs/ciphers.rst:
```diff
@@ -127,15 +124,15 @@ which uses the ChaCha and Poly1305 algorithms, and which is already
 relevant proposals for changing priorities to the Mozilla security
-team first, before asking the Let's Encrypt project to change the
-client's priorities. The Mozilla security team is likely to have more
+team first, before asking the Let's Encrypt project or EFF to change
```
This should just be "the Certbot developers" (rather than "the Let's Encrypt project or EFF").

schoen commented on docs/ciphers.rst:
```diff
@@ -172,23 +169,22 @@ This will probably look something like
+  certbot --update-ciphers off
 
 to disable automatic configuration updates. These features have not yet
 been implemented and this syntax may change then they are implemented.
```
Should be "when they are implemented".

schoen commented on docs/using.rst:
```diff
@@ -525,19 +530,19 @@ whole process is described in the :doc:`contributing`.
+Unless you have a very specific requirements, we kindly suggest that you use
+the certbot-auto_ method. It's the fastest, the most thoroughly
 tested and the most reliable way of getting our software and the free
 SSL certificates!
```
Should be "TLS/SSL certificates", "TLS certificates", or "certificates".

schoen commented:
I just added four comments and am otherwise fine with this.

schoen commented:
LGTM, still curious about the `letsencrypt-apache` reference (the documentation still says that the plugin will have the old name) -- is that a renaming-sequence problem?

schoen commented:
LGTM :-)

bmw commented:
LGTM
